### PR TITLE
Squash wagtailpages migrations

### PR DESCRIPTION
**Pull request by Pomax**
Partially addresses https://github.com/mozilla/foundation.mozilla.org/issues/6510 

This PR is part one of migration squashing, cleaning up the wagtail migrations, with all "intermediate runPython" code that was necessary to move between two specific migrations removed. The only migration data bootstrapping left is the code that creates the blog categories, "area of focus" entries, and PNI product categories, all updated to include a `locale_id` as the squashed migrations now immediately set up a schema with wagtail-localize enabled.

mozfest migrations will be handled in a separate PR, as migration work tends to cause CI to do really weird things, and I'd rather land smaller PRs that pass than keep trying to do this as one giant PR that doesn't =)

Testing is mostly a matter of running checking out this branch, (optionally running `inv catchup` if you've been working on branches with updated/different dependencies, then) running `inv test` and making sure that doesn't report errors for you. If there are none: see what happens on an `inv copy-prod-db` followed by `docker-compose up` by poking around in the site locally (both content and CMS) to see if it still behaves the way you think it should be behaving.

### CODE REVIEW

**Comment by tbrlpld** [comment]
@Pomax Rebuilding is failing for me on the migration step:

```
Operations to perform:
  Apply all migrations: admin, auth, buyersguide, campaign, contenttypes, highlights, mozfest, news, people, redirects, sessions, sites, social_django, taggit, wagtail_footnotes, wagtail_localize, wagtail_localize_git, wagtailadmin, wagtailcore, wagtaildocs, wagtailembeds, wagtailforms, wagtailimages, wagtailinventory, wagtailmedia, wagtailmetadata, wagtailpages, wagtailredirects, wagtailsearch, wagtailusers
Traceback (most recent call last):
  File "/app/network-api/manage.py", line 34, in <module>
    execute_from_command_line(sys.argv)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/base.py", line 371, in execute
    output = self.handle(*args, **options)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/base.py", line 85, in wrapped
    res = handle_func(*args, **kwargs)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/commands/migrate.py", line 202, in handle
    pre_migrate_apps = pre_migrate_state.apps
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/db/migrations/state.py", line 208, in apps
    return StateApps(self.real_apps, self.models)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/db/migrations/state.py", line 278, in __init__
    raise ValueError("\n".join(error.msg for error in errors))
ValueError: The field mozfest.MozfestPrimaryPage.signup was declared with a lazy reference to 'wagtailpages.signup', but app 'wagtailpages' isn't installed.

```

**Review by tbrlpld — COMMENTED** [review]
@Pomax I needed to rebuild the environment with `inv new-env`. `inv catchup` had failed. With the rebuild all is good. The tests pass. I have clicked around the site with production data and all seems to work.

I am a little concerned that this PR already removes the existing migrations. It is recommended to only add the squash migrations and only remove the existing migrations once the squashed migrations are deployed to all environments. https://docs.djangoproject.com/en/4.0/topics/migrations/#migration-squashing
Is there a reason we are diverting from that practice? I am wondering if that might have caused the issue for me where `inv catchup` was not working.

File: requirements.in [hunk 1]
```diff
@@ -23,7 +23,7 @@ python-slugify
 requests
 social-auth-app-django
 wagtail-airtable
-wagtail==2.15.1
+wagtail==2.15.3
```
Comment (Pomax) [inline requirements.in:26]:
```suggestion
wagtail==2.15.1
```

Comment (Pomax) [inline requirements.in:26]: patch version update for wagtail ensures that the "dependencies" entries don't lead to "missing migrations" errors

**Review by danielfmiranda — COMMENTED** [review]
Hi Pomax, while following the instructions, it appears that pulling this branch and running`inv catchup` +`inv test` seems to work well. No errors to report!

However when running `inv copy-prod-db`, I seem to run into some migration related issues. Below is a copu of what error is produced, even after double checking I am logged into the heroku CLI, etc:
```

File "/app/network-api/manage.py", line 34, in <module>
    execute_from_command_line(sys.argv)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/base.py", line 371, in execute
    output = self.handle(*args, **options)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/base.py", line 85, in wrapped
    res = handle_func(*args, **kwargs)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/core/management/commands/migrate.py", line 95, in handle
    executor.loader.check_consistent_history(connection)
  File "/app/dockerpythonvenv/lib/python3.9/site-packages/django/db/migrations/loader.py", line 302, in check_consistent_history
    raise InconsistentMigrationHistory(
django.db.migrations.exceptions.InconsistentMigrationHistory: Migration mozfest.0001_to_0015 is applied before its dependency wagtailcore.0066_collection_management_permissions on database 'default'.
```

File: network-api/networkapi/mozfest/migrations/0001_to_0015.py [hunk 2]
```diff
@@ -28,9 +34,9 @@ class Migration(migrations.Migration):
     initial = True
 
     dependencies = [
-        ('wagtailcore', '0045_assign_unlock_grouppagepermission'),
-        ('wagtailimages', '0001_squashed_0021'),
-        ('wagtailpages', '0001_initial'),
+        ('wagtailcore', '0066_collection_management_permissions'),
+        ('wagtailimages', '0023_add_choose_permissions'),
```
Comment (Pomax) [inline network-api/networkapi/mozfest/migrations/0001_to_0015.py:38]:
```suggestion
        ('wagtailcore', '0045_assign_unlock_grouppagepermission'),
        ('wagtailimages', '0001_squashed_0021'),
```

**Comment by Pomax** [comment]
@danielfmiranda I've rolled back the mozfest migration update a little, can you check if it still gives you that error now?

**Review by fessehaye — CHANGES_REQUESTED** [review]
```
..................................
======================================================================
FAIL: test_no_migrations_missing (network-api.networkapi.tests.MissingMigrationsTests)
Ensure we didn't forget a migration
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/network-api/networkapi/tests.py", line 64, in test_no_migrations_missing
    raise AssertionError("Missing migrations detected:\n" + output.getvalue())
AssertionError: Missing migrations detected:
Migrations for 'mozfest':
  network-api/networkapi/mozfest/migrations/0039_auto_20220208_0612.py
    - Alter field body on mozfestprimarypage
Migrations for 'wagtailpages':
  network-api/networkapi/wagtailpages/migrations/0003_auto_20220208_0612.py
    - Alter field body on modularpage
    - Alter field body on primarypage


----------------------------------------------------------------------
Ran 82 tests in 130.072s

FAILED (failures=1, skipped=2)
Destroying test database for alias 'default'...
1
```

**Comment by tbrlpld** [comment]
@Pomax All is well for me 👍

**Review by tbrlpld — APPROVED** [review]

**Comment by Pomax** [comment]
closing this attempt, let's do mozfest first, make sure that works, and then we can redo wagtailpages with one less potential bottleneck in the way

**Pull request closed without merging**